src/Compilers/Core/Portable/Emit/EmitOptions.cs

```diff
@@ -245,7 +267,7 @@ internal void ValidateOptions(DiagnosticBag diagnostics, CommonMessageProvider m
             }
         }
 
-        internal bool EmitDynamicAnalysisData => !string.IsNullOrEmpty(Instrument);
+        internal bool EmitDynamicAnalysisData => !InstrumentationKinds.IsDefaultOrEmpty && InstrumentationKinds.Contains(InstrumentationKind.TestCoverage);
```

Comment:
If I understood correctly, it seems that `InstrumentationKinds` cannot be default (it could either be empty or be filled).